tests/02_00_02-2wp_segwit_compatible.js

```diff
@@ -1,74 +1,57 @@
+const { sendPegin, MINIMUM_PEGIN_VALUE_IN_BTC, ensurePeginIsRegistered } = require('../lib/2wp-utils');
+const { getBridge, getLatestActiveForkName } = require('../lib/precompiled-abi-forks-util');
+const { getBtcClient } = require('../lib/btc-client-provider');
+const { getDerivedRSKAddressInformation } = require('@rsksmart/btc-rsk-derivation');
+const btcEthUnitConverter = require('btc-eth-unit-converter');
+
 
-var federationAddress;
-var btcClient;
-var rskClient, rskClients;
-var pegClient;
-var test;
 let rskTxHelpers;
+let rskTxHelper;
+let btcTxHelper;
+
 
-const NETWORK = bitcoin.networks.testnet;
+const fulfillRequirementsToRunAsSingleTestFile = async () => {
+    await rskUtils.activateFork(Runners.common.forks.fingerroot500);
+};
 
 describe('Lock using p2sh-p2wpkh address', () => {
     before(async () => {
-        btcClient = bitcoin.getClient(
-            Runners.hosts.bitcoin.rpcHost,
-            Runners.hosts.bitcoin.rpcUser,
-            Runners.hosts.bitcoin.rpcPassword,
-            NETWORK
-        );
-        rskClient = rsk.getClient(Runners.hosts.federate.host);
-        rskClients = Runners.hosts.federates.map(federate => rsk.getClient(federate.host));
-        pegClient = pegUtils.using(btcClient, rskClient);
-        test = pegAssertions.with(btcClient, rskClient, pegClient);
-        utils = rskUtilsLegacy.with(btcClient, rskClient, pegClient);
         rskTxHelpers = getRskTransactionHelpers();
+        rskTxHelper = rskTxHelpers[0];
+        btcTxHelper = getBtcClient();
 
-        // Grab the federation address
-        federationAddress = await rskClient.rsk.bridge.methods.getFederationAddress().call();
-        await btcClient.importAddress(federationAddress, 'federations');
-        
-        // Mine a few rsk blocks to prevent being at the beginning of the chain,
-        // which could trigger border cases we're not interested in
-        await sequentialPromise(10, () => rskUtils.mineAndSync(rskTxHelpers));
-        
-        // Update the bridge to sync btc blockchains
-        await rskClient.fed.updateBridge();
-        await rskUtils.mineAndSync(rskTxHelpers);
-
-        return federationAddress;
+        if(process.env.RUNNING_SINGLE_TEST_FILE) {
+            await fulfillRequirementsToRunAsSingleTestFile();
+        }
     });
 
-    it('should work when using p2sh-p2wpkh address', async () => {
-        try {
-            const INITIAL_BTC_BALANCE = bitcoin.btcToSatoshis(40);
-            const INITIAL_RSK_BALANCE = bitcoin.btcToSatoshis(10);
-            const MAX_EXPECTED_FEE = bitcoin.btcToSatoshis(0.001);
+    it('should do a legacy pegin using p2sh-p2wpkh address', async () => {
+        const senderAddressInfo = await btcTxHelper.generateBtcAddress('p2sh-segwit');
 
-            var addresses = await pegClient.generateNewAddress('test', 'p2sh-segwit');
-            expect(addresses.inRSK).to.be.true;
+        const bridge = getBridge(rskTxHelper.getClient(), await getLatestActiveForkName());
```

Comment:
```suggestion
        const latestActiveForkName = await getLatestActiveForkName();
        const bridge = getBridge(rskTxHelper.getClient(), latestActiveForkName);
```

Reply:
why? what is the advantage pls?

Reply:
It's Bad Practise Passing An Async Function As Parameter To Another Function. See Error That It Could Lead To [Here](https://plainenglish.io/blog/be-careful-of-passing-an-async-function-as-a-parameter-4f421f8e7e9d)

Reply:
thanks for the explanation!